ntbs-service/Services/AzureAdImportService.cs

```diff
@@ -24,12 +25,23 @@ public async Task RunCaseManagerImportAsync()
             var tbServices = await _referenceDataRepository.GetAllTbServicesAsync();
             using (var azureAdDirectoryService = _azureAdDirectoryServiceFactory.Create())
             {
-                var users = await azureAdDirectoryService.LookupUsers(tbServices);
+                var users = (await azureAdDirectoryService.LookupUsers(tbServices)).ToList();
                 foreach (var (user, tbServicesMatchingGroups) in users)
                 {
                     Log.Information($"Updating user {user.Username}");
                     await _userRepository.AddOrUpdateUser(user, tbServicesMatchingGroups);
                 }
+
+                var ntbsUsersNotInAd = this._userRepository.GetUserIQueryable()
+                    .Where(user => !users.Select(u => u.user.Username)
+                        .Contains(user.Username)).ToList();
+                foreach (var user in ntbsUsersNotInAd)
+                {
+                    Log.Information($"Updating user {user.Username}");
+                    user.IsActive = false;
+                    user.AdGroups = null;
+                    await _userRepository.AddOrUpdateUser(user, user.CaseManagerTbServices.Select(cmtb => cmtb.TbService));
+                }
             }
```

Comment:
This logic is very similar to that above. Could we commonise it?

Reply:
I have now commonised to use 1 foreach loop